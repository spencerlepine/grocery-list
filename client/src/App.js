import React from "react";
import { useSelector } from "react-redux";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import NavBar from "./components/NavBar/NavBar";
import BrowseGroceries from "./components/BrowseGroceries/BrowseGroceries";
import Form from "./components/Form/Form";
import Cart from "./components/Cart/Cart";
import Settings from "./components/Settings/Settings";
import Error from "./components/Error/Error";
import Footer from "./components/Footer/Footer";
import Login from "./components/Login/Login";
import SignUp from "./components/SignUp/SignUp";
import Recommended from "./components/Recommended/Reccomended";
import Pantry from "./components/Pantry/Pantry"
import NotificationPopup from "./components/NotificationPopup/NotificationPopup"
import FetchLogic from "./components/FetchLogic/FetchLogic"
import "./index.css";

const App = () => {
  const userId = useSelector(state => state.connectedUser)

  return (
    <BrowserRouter>
      <NotificationPopup />
      {userId ? (<>
        <NavBar />
        <FetchLogic />
        <Switch>
          <Route exact path="/">
            <BrowseGroceries />
          </Route>
          <Route path="/pantry">
            <Pantry />
          </Route>
          <Route path="/form">
            <Form />
          </Route>
          <Route path="/cart">
            <Cart />
          </Route>
          <Route path="/settings">
            <Settings />
          </Route>
          <Route path="/recommended">
            <Recommended />
          </Route>
          <Route path="/">
            <Error />
          </Route>
        </Switch>
      </>) : (
        <Switch>
          <Route exact path="/register">
            <SignUp />
          </Route>
          <Route path="/">
            <Login />
          </Route>
        </Switch>
      )}
      <Footer />
    </BrowserRouter>
  );
};

export default App;
