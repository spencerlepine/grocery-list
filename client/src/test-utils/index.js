import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { AuthContext } from 'context/AuthContext/AuthContext';
import { CartProvider } from 'context/CartContext/CartContext';
import { FormProvider } from 'context/FormContext/FormContext';
import { NotificationsProvider } from 'context/NotificationsContext/NotificationsContext';
import { ProductsProvider } from 'context/ProductsContext/ProductsContext';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import currentUser from 'currentUser';
import '@testing-library/jest-dom';

jest.mock('config/firebase', () => ({
  auth: null,
  db: null,
}));

window.toTitleCase = str => {
  const result = str.replace(/([A-Z])/g, ' $1');
  return result.charAt(0).toUpperCase() + result.slice(1);
};

const history = createBrowserHistory();

const AllTheProviders = ({ children }) => (
  // <ThemeProvider theme="light">
  <AuthContext.Provider value={{ currentUser }}>
    <NotificationsProvider>
      <ProductsProvider>
        <CartProvider>
          <FormProvider>
            <Router history={history}>
              {children}
            </Router>
          </FormProvider>
        </CartProvider>
      </ProductsProvider>
    </NotificationsProvider>
  </AuthContext.Provider>
  // </ThemeProvider>
);

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });


// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
