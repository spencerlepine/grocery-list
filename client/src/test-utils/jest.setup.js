const CONSOLE_FAIL_TYPES = ['error', 'warn']

CONSOLE_FAIL_TYPES.forEach((type) => {
  console[type] = (message) => (
    `Failing due to console.${type} while running test!\n\n${message}`
  )
})