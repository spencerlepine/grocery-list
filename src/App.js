import React from "react";
import Navbar from "./components/Navbar/Navbar";
import Layout from "./wrappers/Layout/Layout";
import Footer from "./components/Footer/Footer";
import Routes from "./wrappers/Routes/Routes";
import NotificationsPopup from "./components/NotificationsPopup/NotificationsPopup";

const App = () => (
  <>
    <NotificationsPopup />
    <Navbar />
    <Layout>
      <Routes />
    </Layout>
    <Footer />
  </>
);

export default App;
