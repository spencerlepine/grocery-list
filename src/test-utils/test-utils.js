import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
// import { ThemeProvider } from 'my-ui-lib';
import { AuthProvider } from '../context/AuthContext/AuthContext';
import { CartProvider } from '../context/CartContext/CartContext';
import { FormProvider } from '../context/FormContext/FormContext';
import { NotificationsProvider } from '../context/NotificationsContext/NotificationsContext';
import { ProductsProvider } from '../context/ProductsContext/ProductsContext';

const AllTheProviders = ({ children }) => (
  // <ThemeProvider theme="light">
  <CartProvider>
    <FormProvider>
      <NotificationsProvider>
        <ProductsProvider>
          <AuthProvider>
            {children}
          </AuthProvider>
        </ProductsProvider>
      </NotificationsProvider>
    </FormProvider>
  </CartProvider>
  // </ThemeProvider>
);

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });


// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
