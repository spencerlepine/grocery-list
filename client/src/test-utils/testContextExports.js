import React from 'react';
import { screen, render } from 'test-utils';

const testContextExports = (ContextProdiver, useContext, expectedExports) => {
  it('expectedExports not be an empty array', () => {
    expect(expectedExports instanceof Array).toBeTruthy();
    expect(expectedExports.length).toBeGreaterThan(0);
  });

  it('expectedExports should contain objects', () => {
    const validElements = expectedExports.map(elem => elem instanceof Object);
    expect(validElements.every(e => e === true)).toBeTruthy();
  });

  it('expectedExports should contain "key" key with string value', () => {
    const firstElem = expectedExports[0];

    expect(firstElem['key']).not.toBe(undefined);
    expect(firstElem['key'].constructor === String).toBeTruthy();
  });

  it('expectedExports should contain "targetInstance" key with contructor value', () => {
    const firstElem = expectedExports[0];

    expect(firstElem['targetInstance']).not.toBe(undefined);
    expect(firstElem['targetInstance'] instanceof Object).toBeTruthy();
  });

  it('useContext should be a function', () => {
    expect(useContext instanceof Function).toBeTruthy();
  });

  it('should render ContextProdiver with no errors', () => {
    expect(() => render(<ContextProdiver><p>MOCK CHILD</p></ContextProdiver>)).not.toThrow(new Error);
  });

  it('should render the children', () => {
    render(<ContextProdiver><p data-testid="mockChild">MOCK CHILD</p></ContextProdiver>);

    expect(screen.getByTestId('mockChild')).toBeInTheDocument();
    expect(screen.getByText('MOCK CHILD')).toBeInTheDocument();
  });

  test('passes correct values to children', () => {
    const MockChild = () => {
      const productExports = useContext();

      expectedExports.forEach(({ key, targetInstance }) => {
        try {
          expect(productExports[key]).not.toBe(undefined);
        } catch {
          throw new Error(`expected value ${productExports[key]} (at key: ${key}) NOT to be undefined`);
        }

        try {
          expect(productExports[key].constructor === targetInstance).toBeTruthy();
        } catch {
          throw new Error(`expected ${productExports[key]} (at key: ${key}) to be instance of ${targetInstance.toString()}`);
        }
      });

      return <p>test</p>;
    };

    render(<ContextProdiver><MockChild /></ContextProdiver>);
  });
};

/* eslint-disable */
export default testContextExports;
