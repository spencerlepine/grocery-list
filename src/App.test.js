import React from 'react';
import { render, cleanup } from './test-utils';
import App from './App';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<App />);

  expect(asFragment(<App />)).toMatchSnapshot();
});
