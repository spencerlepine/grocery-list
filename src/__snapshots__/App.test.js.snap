// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="notifications-popup makeStyles-popupContainer-4"
  >
    <p>
      under construction
    </p>
  </div>
  <nav
    class="navbar makeStyles-navbar-7"
  >
    <div
      class="navbarContainer makeStyles-navbarContainer-6"
    >
      <div
        class="sidebar undefined"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-toggleIcon-9 MuiSvgIcon-fontSizeLarge"
          focusable="false"
          style="visibility: visible;"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </div>
      <a
        class="makeStyles-logoLink-8"
        href="/"
      >
        <img
          alt="QuickCart Logo"
          class="makeStyles-logoLink-8"
          src="QuickCart-Logo.png"
        />
      </a>
      <a
        class="makeStyles-cartLink-14"
        href="/cart"
      >
        <div
          class="makeStyles-cartLink-14"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </div>
      </a>
    </div>
  </nav>
  <div
    class="makeStyles-navbarSpacing-2"
  />
  <div
    class="makeStyles-content-3"
  >
    <div
      class="welcome-container makeStyles-welcomeContainer-20"
    >
      <p
        class="makeStyles-appDescription-26"
      >
        Organize and budget a personal grocery cart for easy shopping
      </p>
      <div
        class="makeStyles-accountRedirect-25"
      >
        <a
          class="makeStyles-accountLink-22"
          href="/signup"
        >
          Sign up
        </a>
        <a
          class="makeStyles-loginLink-23"
          href="/login"
        >
          Log in
        </a>
      </div>
      <img
        alt="QuickCart Screenshot"
        class="makeStyles-demoImg-21"
        src="demo_transparent.png"
      />
      <div>
        <h2>
          Search Products
        </h2>
        <img
          alt="search products on QuickCart"
          src=""
        />
      </div>
      <div>
        <h2>
          Generate a Shopping List
        </h2>
        <img
          alt="generate a shopping list on QuickCart"
          src=""
        />
      </div>
      <div>
        <h2>
          Customize Product Details
        </h2>
        <img
          alt="customize product details on QuickCart"
          src=""
        />
      </div>
      <div>
        <h2>
          Create an Account Today!
        </h2>
        <a
          class="makeStyles-accountLink-22"
          href="/signup"
        >
          Sign up
        </a>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-footerSpacing-1"
  />
  <footer
    class="footer makeStyles-footer-16"
  >
    <div
      class="makeStyles-copyright-17"
    >
      <p>
        Built by Spencer Lepine
      </p>
    </div>
    <div
      class="makeStyles-socials-18"
    >
      <a
        class="makeStyles-socialsLink-19"
        href="https://twitter.com/SpencerLepine"
        rel="noopener noreferrer"
        target="_blank"
      >
        <p>
          Twitter
        </p>
      </a>
      <a
        class="makeStyles-socialsLink-19"
        href="https://github.com/spencerlepine"
        rel="noopener noreferrer"
        target="_blank"
      >
        <p>
          GitHub
        </p>
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
